protocols/autonat/src/behaviour/as_server.rs

```diff
@@ -295,8 +295,8 @@ impl<'a> AsServer<'a> {
             .values()
             .find_map(|a| a.as_ref())
             .ok_or_else(|| {
-                let status_text = "no dial-request over relayed connections".to_string();
-                (status_text, ResponseError::DialError)
+                let status_text = "refusing to dial peer with blocked observed address".to_string();
+                (status_text, ResponseError::DialRefused)
```

Comment:
Note: the go implementation returns a `ResponseError::DialError` here (but with the same status text). I decided to change it here because imo a `DialRefused` makes more sense. A client should not flip their status to private or reduce the confidence in their public status just because they accidentally picked a server that rejects them because of  the observed address (though if the client also enabled `only_global_ips` this should not happen in practice).

Reply:
>  Edit: should it be part of the spec to list the cases in which a server returns DialError?

Yes. I think this is worth consolidating on. Mind driving it on libp2p/specs?

Reply:
Opened https://github.com/libp2p/specs/issues/411#issue-1231639520.

Reply:
Also added 7035422, which does the same change for the case that `filter_valid_addrs` filters out all addresses send from the client.